### PR TITLE
Simplifies the way burning food is handled, kills burns_in_oven and burns_on_grill

### LINKED ISSUE #77377: Baked boiled rice has no after-cook time
Reporting client version: 514.1589

## Round ID:
[211744](https://scrubby.melonmesa.com/round/211744)

## Testmerges:
- [Feature: bitrunner, a new supply role](https://www.github.com/tgstation/tgstation/pull/77259)

## Reproduction:
image
I swear there was a very similar issue with baked bell peppers back when they were introduced that I could point at to be like "same hat" but I can't find it

### PR DESCRIPTION
**Pull request by FernandoJ8**
## About The Pull Request

Currently, burning food is handled in the following way (for the sake of brevity I'll explain it with baking but it's essentially the exact same with grilling):
- If a recipe has a baking recipe added in it's `make_bakeable`, then it will produce that when baked.
- If it does not have a recipe added, but `burns_in_oven` is TRUE, then it will have a negative baking recipe for "burned mess" added by the parent of `make_bakeable`, and be turned to that after 20 to 25 seconds.
- If it has neither, it will instantly be set on fire.

What this PR does is, basically, remove the check for `burns_in_oven/burns_on_grill` and make food always have a negative burned mess recipe unless overriden by an actual recipe.

Also modified the examines a bit to better fit food that hasn't necessarily been baked/grilled before but would still turn to a burnt mess if put in either.

## Why It's Good For The Game

My issues with the way it currently works are:

1. The "default" case is one that largely doesn't make sense for food. Most things would logically just get turned to a burnt mess after being overcooked for some time rather than combusting the very moment you close the range.
2. If you make any cooking item the result of a baking recipe, then you must also remember to make its `burns_in_oven` var true, because otherwise it will be set on fire as soon as it finishes baking, making it impossible to cook. (See issue #77377) This is pretty bad because it's easy not to notice or add these vars, given that they're a bit out of the way mechanically and...
3. ...the var names are very unintuitive. In practical terms all `burns_in_oven` and `burns_in_grill` do is add a timer before the item is ruined, not actually cause or prevent the ruining.

By removing these vars and their function we deal with all three, as turning into a burnt mess is now the default outcome for food that shouldn't be baked/grilled and coders adding food/recipes don't need to worry about the confusingly-named vars.

Also, as mentioned before, fixes #77377.

## Changelog
:cl:
fix: boiled rice doesn't burn instantly after being baked
code: simplifies the way burning food is handled, grilled/baked food now turns to a burnt mess rather than being set on fire, unless they have a baking/grilling recipe
/:cl:

### CODE REVIEW

File: code/datums/components/grillable.dm
Comment (AdipemDragon) [inline code/datums/components/grillable.dm:1]: Shouldn't it be "on the grill"?

**Comment by san7890** [comment]
will clash with #77562

File: code/datums/components/grillable.dm
Comment (FernandoJ8) [reply code/datums/components/grillable.dm:1]: Ah, prepositions.

**Comment by FernandoJ8** [comment]
> will clash with #77562

Do that one first, I suppose, since it's the one compliant with current code as opposed to altering it.

**Comment by san7890** [comment]
undraft when all fires relating to clashing pr is resolved

**Comment by SyncIt21** [comment]
Fixes #77807

**Comment by SyncIt21** [comment]
@FernandoJ8 this PR ready?

**Comment by Ghommie** [comment]
Lots of merge conflicts.

**Comment by FernandoJ8** [comment]
> Lots of merge conflicts.



> Lots of merge conflicts.

Should be ready if/when it passes checks.

**Review by Ghommie — APPROVED** [review]
Oh well, this means every food item now has these two components attached, but I don't think it's a big deal overall.

**Pull request merged**